File: Problem_of_the_Day/SingleNum-III.cpp
class Solution {
public:
    vector<int> singleNumber(vector<int>& nums) {
        int sz=nums.size();
        long long remove_dup=0;
        
        for(auto i:nums){
            remove_dup^=i;
        }
        long long sb=(remove_dup)&(remove_dup-1);
        sb^=remove_dup;
        
        int one=0;
        int zero=0;
        for(auto i:nums){
            if(i&sb){
                one^=i;
            }
            else{
                zero^=i;
            }
        }
        vector<int> answer={one,zero};
        return answer;
    }
};

File: Problem_of_the_Day/SingleNum-III.py
class Solution:
    def singleNumber(self, nums: List[int]) -> List[int]:
        n = len(nums)
        x = 0
        for i in nums:
            x^=i;
        sb = x & (x-1)
        sb ^= x
        one = 0
        zero = 0
        for i in nums:
            if i & sb:
                one ^= i
            else:
                zero ^= i
        answer = []   
        answer.append(one)
        answer.append(zero)
        return answer
        
        
  
